### PR TITLE
FORMS-3696 adding rich text support for labels

### PR DESCRIPTION
**Pull request by im-shiv**

### CODE REVIEW

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/base/v1/base/clientlibs/editor/js/editDialog.js [hunk 1]
```diff
@@ -193,4 +257,15 @@
         }
     });
 
+    /**
+     * whenever RTE value of richTextTitle is changed, we save it's value to corresponding text value field.
+     */
+    channel.on("change", "div[name='richTextTitle']", function (e) {
```
Comment (rismehta) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/base/v1/base/clientlibs/editor/js/editDialog.js:263]: Please use BEM for richTextTitle

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/base/v1/base/clientlibs/editor/js/editDialog.js [hunk 2]
```diff
@@ -164,14 +166,70 @@
...
+     * @param isToggled is this function being called on toggle of rich text checkbox.
+     */
+    function resolveRichText(dialog, isTitleRichText, isToggled) {
+        var title = dialog.find(BASE_TITLE)[0],
+            richTextTitle = dialog.find(BASE_RICH_TEXT_TITLE)[0],
+            richTextTitleDiv = dialog.find("div[name='richTextTitle']")[0];
+        if(isTitleRichText.checked){
+            hideGraniteComponent(title);
+            showGraniteComponent(richTextTitle);
+            showGraniteComponent($(richTextTitle).parent('div.richtext-container').siblings('label'));
```
Comment (rismehta) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/base/v1/base/clientlibs/editor/js/editDialog.js:184]: All these should follow BEM. No CSS selector without BEM paradigm

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/af-commons/v1/fieldTemplates/label.html [hunk 3]
```diff
@@ -1,4 +1,4 @@
-<template data-sly-template.label="${@ componentId, labelValue, labelVisible, bemBlock}">
+<template data-sly-template.label="${@ componentId, labelValue, labelVisible, labelRichText, bemBlock}">
     <label for="${componentId}" class="${bemBlock}__label"
-           data-sly-test="${labelValue && (labelVisible==null || labelVisible==true)}">${labelValue}</label>
+           data-sly-test="${labelValue && (labelVisible==null || labelVisible==true)}" isTitleRichText="${labelRichText}">${labelValue @ context = labelRichText ? 'html' : 'text'}</label>
```
Comment (rismehta) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/af-commons/v1/fieldTemplates/label.html:3]: why do you need isTitleRichText ? if you need it to access in JS, it should have a data attribute. HTML elements should either have BEM for CSS or data attributes for JS interactions

**Review by rismehta — CHANGES_REQUESTED** [review]
-1 for now

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/af-commons/v1/fieldTemplates/label.html (hunk 3 above)
Comment (im-shiv) [reply ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/af-commons/v1/fieldTemplates/label.html]: agreed, changed to data-richtext attribute for interaction with js

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/base/v1/base/clientlibs/editor/js/editDialog.js (hunk 2 above)
Comment (im-shiv) [reply ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/base/v1/base/clientlibs/editor/js/editDialog.js]: removed it

**Pull request merged**